packages/application-shell/src/configure-apollo.ts

```diff
@@ -100,22 +100,9 @@ export const createApolloClient = () =>
       fragmentMatcher,
       dataIdFromObject: result => {
         const customResult = result as CustomIdGetterObj;
-        if (
-          !customResult.id &&
-          customResult.key &&
-          customResult.__typename &&
-          typeNamesWithoutIdAsIdentifier.includes(customResult.__typename)
-        )
-          return `${customResult.__typename}:${customResult.key}`;
-        // Generally all id's are unique accross the platform, so we don't need to
-        // include the type name in the cache key.
-        // However, a reference has the shape { typeId, id } where the id is the
-        // id of the referenced entity. If we didn't prefix ids of References
-        // they would clash with the referenced resource.
-        if (customResult.__typename === 'Reference')
```

Comment:
So "all" we do is to always prefix with typename right? Do we anyhow now why we didn't do this in the first place? /cc @emmenko maybe?

Reply:
Can't remember exactly. Some old PRs I found:
- https://github.com/commercetools/merchant-center-application-kit/pull/310
- https://github.com/commercetools/merchant-center-application-kit/commit/072852f434649c92f50c0801b0fa265ebf537ef7
- https://github.com/commercetools/merchant-center-application-kit/commit/20731f6a10e293fdfde23c26d610f57d97422df6